### PR TITLE
Generalize types/formats and integrate Graph Formats

### PR DESCRIPTION
**Pull request by danlamanna**
Still a work in progress. 

At a minimum:
- [x] Add quickview/basic visualization for graphs   
- [x] Add graph extensions (to romanesco)

### CODE REVIEW

**Comment by danlamanna** [comment]
@jeffbaumes This is ready for review - the only thing missing is a method for sampling graphs to ease the visual display. This should probably be done on the Romanesco side of things, let me know what you think.

File: app/src/views/DatasetManagementView.js [hunk 1]
```diff
@@ -54,12 +29,26 @@
...
                         nameWithExtension = parts[parts.length - 1] === extension ? name : name + '.' + extension,
-                    file = flow.girderUpload(blob, nameWithExtension, flow.saveLocation.get('dataFolder'), false, function () {
-                        dataset.set({id: file.get('itemId')});
-                    });
+                        file = flow.girderUpload(blob, nameWithExtension, flow.saveLocation.get('dataFolder'), false, function () {
+                            // Save type/format on the item for retrievel later
+                            // These get set directly on the dataset model
+                            var item = new girder.models.ItemModel({
+                                _id: file.get('itemId')
+                            }).fetch();
```
Comment (danlamanna) [inline app/src/views/DatasetManagementView.js]: Is this the typical way of retrieving the item model from an id?

Comment (zachmullen) [inline app/src/views/DatasetManagementView.js]: Yep

File: app/src/views/DatasetManagementView.js [hunk 2]
```diff
@@ -54,12 +29,26 @@
...
+
+                            // _.noop Can be undefined with girder/girder#1057
+                            item.addMetadata('flow', {
+                                type: dataset.get('type'),
+                                format: dataset.get('format')
+                            }, _.noop, function () {
+                                window.flow.bootstrapAlert("danger", "Failed to set metadata on item");
+                            });
```
Comment (danlamanna) [inline app/src/views/DatasetManagementView.js]: The file upload and the metadata set should both be treated atomically. Perhaps using something like the facilities mentioned in girder/girder#1072.

**Comment by danlamanna** [comment]
@jeffbaumes Let me know what you think.

File: app/src/templates/prompt.jade [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            button.close(type="button", data-dismiss="modal")
+                span(aria-hidden="true")
+                    | &times;
+                span.sr-only
+                    | Close
+            h4.modal-title
+                | #{prompt}
+        .modal-body
+            .form-inline
+              select(id="prompt-choice")
```
Comment (jeffbaumes) [inline app/src/templates/prompt.jade]: I think this will look better as a bootstrap-themed `select.form-control`

**Comment by jeffbaumes** [comment]
LGTM :snowflake:

**Pull request merged**